### PR TITLE
feat: hash with SHA256 cross-platform

### LINKED ISSUE #1084: useResponseCache crypto module usage
The response cache plugin currently uses the crypto module for generating hashes which is only available on Node.js. For true cross platform we should investigate alternatives.

### PR DESCRIPTION
**Pull request by n1ru4l**
Closes https://github.com/dotansimha/envelop/issues/1084

### CODE REVIEW

File: .changeset/chilled-donuts-search.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+'@envelop/response-cache': minor
```
Comment (dotansimha) [inline .changeset/chilled-donuts-search.md:2]: @n1ru4l should be major?

**Review by dotansimha — APPROVED** [review]
LGTM. @n1ru4l please check my comment on the changeset

File: .changeset/chilled-donuts-search.md (hunk 1 above)
Comment (n1ru4l) [reply .changeset/chilled-donuts-search.md:2]: pre `1.x.x` a minor bump is actually considered a major as long as the package is unstable (see https://semver.org/#spec-item-4)

I am sure we will do a few more breaking changes to the response cache, before making it a full v1. We still need a bit more feedback from the community.

Comment (dotansimha) [reply .changeset/chilled-donuts-search.md:2]: Oh, i see, makes sense, thanks!

**Comment by n1ru4l** [comment]
@dotansimha I haven't tested this on deno or cloudflare workers. We definitely need some kind of automated strategy for doing this. However, the code runs just fine on Node.js and the browser.

**Pull request merged**